lib/trip_dispatcher.rb

```diff
@@ -9,33 +9,94 @@ class TripDispatcher
     attr_reader :drivers, :passengers, :trips
 
     def initialize(directory: './support')
-      @passengers = Passenger.load_all(directory: directory)
-      @trips = Trip.load_all(directory: directory)
+      @passengers = Passenger.load_all(directory: directory) 
+      @trips      = Trip.load_all(directory: directory) 
+      @drivers    = Driver.load_all(directory: directory)
       connect_trips
     end
 
-    def find_passenger(id)
+    def find_passenger(id) 
       Passenger.validate_id(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
 
+    def find_driver(id)
+      Driver.validate_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
     def inspect
-      # Make puts output more useful
       return "#<#{self.class.name}:0x#{object_id.to_s(16)} \
               #{trips.count} trips, \
               #{drivers.count} drivers, \
               #{passengers.count} passengers>"
     end
 
+    def request_trip(passenger_id)
+      passenger = find_passenger(passenger_id)
+      raise ArgumentError, "Passenger id not found" if passenger == nil
+
+      driver = @drivers.select {|driver| driver.status == :AVAILABLE}.first
```

Comment:
You can use `find` to make this a little clearer:

```suggestion
      driver = @drivers.find {|driver| driver.status == :AVAILABLE}
```